src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructureUnitTest.java

```diff
@@ -1,19 +1,359 @@
+        final RandomDAG dag = RandomDAG.getRandomDAG();
+        final ComputableGraphStructure cgs = new ComputableGraphStructure(dag.getEquivalentCacheNodeSet());
+        dag.nodeKeysSet.forEach(nodeKey -> Assert.assertTrue(isTopologicallyEquivalent(dag.getTopologicalOrderForNodeMutation(nodeKey),
+                cgs.getTopologicalOrderForNodeMutation(nodeKey), dag.topologicalOrderMap)));
+    }
+
+    @Test(invocationCount = NUM_TRIALS)
+    public void testTopologicalOrderForTagEvaluation() {
+        final RandomDAG dag = RandomDAG.getRandomDAG();
+        final ComputableGraphStructure cgs = new ComputableGraphStructure(dag.getEquivalentCacheNodeSet());
+        dag.tagsSet.forEach(tag -> Assert.assertTrue(isTopologicallyEquivalent(dag.getTopologicalOrderForTagEvaluation(tag),
+                cgs.getTopologicalOrderForTagEvaluation(tag), dag.topologicalOrderMap)));
+    }
+
+    @Test(invocationCount = NUM_TRIALS)
+    public void testTopologicalOrderForCompleteEvaluation() {
+        final RandomDAG dag = RandomDAG.getRandomDAG();
+        final ComputableGraphStructure cgs = new ComputableGraphStructure(dag.getEquivalentCacheNodeSet());
+        final List<String> orderedNodes = new ArrayList<>(dag.nodeKeysSet);
+        orderedNodes.sort(Comparator.comparingInt(dag.topologicalOrderMap::get));
+        Assert.assertTrue(isTopologicallyEquivalent(cgs.getTopologicalOrderForCompleteEvaluation(), orderedNodes,
+                dag.topologicalOrderMap));
+    }
+
+    @Test
+    public void testAssertTopologicallyEquivalentLists() {
+        final Map<String, Integer> topologicalOrderMap = ImmutableMap.<String, Integer>builder()
+                .put("a0", 0).put("b0", 0).put("c0", 0)
+                .put("a1", 1).put("b1", 1).put("c1", 1)
+                .put("a2", 2).put("b2", 2).put("c2", 2).put("d2", 2).build();
+        Assert.assertTrue(isTopologicallyEquivalent(
+                Arrays.asList("a0", "c0", "a2", "d2"),
+                Arrays.asList("c0", "a0", "a2", "d2"),
+                topologicalOrderMap));
+        Assert.assertTrue(isTopologicallyEquivalent(
+                Arrays.asList("a0", "c0", "b1", "a1", "c1", "a2", "d2"),
+                Arrays.asList("c0", "a0", "b1", "c1", "a1", "a2", "d2"),
+                topologicalOrderMap));
+        Assert.assertTrue(!isTopologicallyEquivalent(
+                Arrays.asList("c0", "a2", "d2"),
+                Arrays.asList("c0", "a0", "a2", "d2"),
+                topologicalOrderMap));
+        Assert.assertTrue(!isTopologicallyEquivalent(
+                Arrays.asList("a0", "c0", "b1", "a1", "c1", "a2", "c2"),
+                Arrays.asList("c0", "a0", "b1", "c1", "a1", "a2", "d2"),
+                topologicalOrderMap));
+    }
+
+    /**
+     * This test helper class creates a random DAG starting from a topological order. All helper methods are
+     * implemented in a brute-force manner.
+     */
+    private static final class RandomDAG {
```

Comment:
Let me do this after the beta release -- I'm still hoping to get this PR to merged before beta!